Second round of migration of the schemes

HechtiDerLachs opened the pull request:
This will migrate the `SpecOpen` related stuff out of the experimental section, following the patterns suggested and implemented by @HereAround . 

Note that `SpecOpen` is not really user facing, but an essential building block for what follows for the covered schemes.

HechtiDerLachs commented:
@simonbrandhorst @HereAround : I already marked several TODOs in the code, mostly about missing or incomplete docstrings. Feel free to add something there. I will now turn towards `SpecOpenMor` and `SpecOpenRing`. The next independent step that could be addressed by anyone else in parallel would be `Glueing` and `SimpleGlueing`. Feel free to push to this branch with any changes in that direction.

HechtiDerLachs commented:
On my local machine the tests are running. I think, these failures are not due to us. Maybe @fingolfin knows what's going on?

thofma commented:
This is https://github.com/oscar-system/Singular.jl/issues/601

simonbrandhorst commented:
I am just changing our naming from `ambient_ring` to `ambient_coordinate_ring`. To avoid a merge conflict, I will avoid changing the files `SpecOpen` `Types.jl` and `SpecialTypes.jl`  and leave the alias `ambient_ring` for now.

simonbrandhorst commented:
Please make separate PRs for the other independent stuff to be cleaned up and moved from experimental.

HechtiDerLachs commented:
Alright. Unless someone volunteers to add documentation for this one here, this should be ready to go, then. Seperate PR's for the next steps, then.

HechtiDerLachs commented:
> I am just changing our naming from `ambient_ring` to `ambient_coordinate_ring`. To avoid a merge conflict, I will avoid changing the stuff in the experimental folder and leave the alias `ambient_ring` for now.

If you rebase on this, you should now not run into any problems, anymore. There is only one instance of `ambient_ring` in all the files I have changed, so your work should not really be affected by this PR.

Edit: Same applies to the next steps. `ambient_ring` is about affine schemes which is already migrated. It makes another occurence in the projective schemes, but these are not on our list for migration, now.

simonbrandhorst commented:
in `SpecOpen` there is tons of `ambient_ring`

HechtiDerLachs commented:
> in `SpecOpen` there is tons of `ambient_ring`

Ok, that's true. Sorry, I had only been thinking about the declarations, not the usage. Then let's get this merged quickly and postpone extension of the documentation until later.

simonbrandhorst commented on src/Schemes/OpenInclusion/Morphisms/Types.jl:
```diff
@@ -0,0 +1,25 @@
+# inclusions of one into the other.                                    #
+########################################################################
+
+@attributes mutable struct OpenInclusion{DomainType, CodomainType, PullbackType}<:AbsSpecMor{DomainType, CodomainType, PullbackType, OpenInclusion, Nothing}
```
You could document the type instead. Then `?OpenInclusion` returns something useful.

HereAround reviewed (approved):
Thank you @HechtiDerLachs . Looks good to me.

Of course, more needs to be added (documentation, consistent names taking into account the other corner stones of OSCAR, possibly code improvements etc.). But one step at a time. For now, the content was not changed but only migrated into the main code source...

Hence, @simonbrandhorst, unless I hear differently, I will merge this once all required tests pass.

HechtiDerLachs committed:
Migrate SpecOpen and related functionality.

This moves it out of the experimental section
and into the main code base. No new functionality
was added.
